r/R/dplyr-functions.R

```diff
@@ -58,7 +58,16 @@ nse_funcs$cast <- function(x, target_type, safe = TRUE, ...) {
 }
 
 nse_funcs$is <- function(object, class2) {
+  assert_that(is.string(class2) || inherits(class2, "DataType"))
   if (is.character(class2)) {
+    class2 <- switch(class2,
+      "utf8" = "string",
+      "float16" = "halffloat",
+      "float32" = "float",
+      "boolean" = "bool",
+      "float64" = "double",
+      class2
+    )
```

Comment:
@nealrichardson this canonicalizes type alias strings to match those returned by `DataType$ToString()`. It doesn't handle the ones that take parameters because that would get very complicated. Would this be useful for anywhere else? If so I can break it out into a separate function.

Reply:
This might be useful in the future so I broke it out into a separate function in 6f7c84b